app/queues/notifications/expiration-notification.server.ts

```diff
@@ -0,0 +1,172 @@
+// Third party modules
+import { Worker, Queue, UnrecoverableError } from 'bullmq';
+import dayjs from 'dayjs';
+
+// Internal modules
+import { redis } from '~/lib/redis.server';
+import logger from '~/lib/logger.server';
+import { addNotification } from './notifications.server';
+import { prisma } from '~/db.server';
+
+// Types
+import type { NotificationData } from './notifications.server';
+
+enum RecordType {
+  Certificate = 'certificate',
+  DnsRecord = 'record',
+}
+interface ExpirationStatusPayload {
+  type: RecordType;
+}
+
+// name for the queue
+const expirationNotificationQueueName = 'expiration-notification';
+
+const updateNotificationStatus = (type: RecordType, id: number) => {
+  switch (type) {
+    case RecordType.Certificate:
+      return prisma.certificate.update({
+        where: {
+          id,
+        },
+        data: {
+          lastNotified: new Date(),
+        },
+      });
+    case RecordType.DnsRecord:
+      return prisma.record.update({
+        where: {
+          id,
+        },
+        data: {
+          lastNotified: new Date(),
+        },
+      });
+  }
+};
+
+// fetch records by status based on type
+const getRecordsByExpiration = (type: RecordType) => {
+  switch (type) {
+    case RecordType.Certificate:
+      return prisma.certificate.findMany({
+        where: {
+          validTo: {
+            lte: dayjs().add(1, 'M').toDate(),
+          },
+          OR: [
+            {
+              lastNotified: null,
+            },
+            {
+              lastNotified: dayjs().subtract(30, 'd').toDate(),
```

Comment:
I would suggest adding a constant NOTIFICATION_FREQUENCY, default 7 (days)
```suggestion
              lastNotified:{
                lte: dayjs().subtract(NOTIFICATION_FREQUENCY, 'd').toDate(),
              },
```
The above is just a tip, but I think it should work :)

Reply:
Thank you. I have updated the code to reflect the changes.